src/views/List.jsx

```diff
@@ -1,23 +1,47 @@
+import { useState, useEffect } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
-	const listItems = data.map((data) => {
-		return <ListItem key={data.id} name={data.name} />;
-	});
+	const [currentItems, setCurrentItems] = useState('');
+	const [list, setList] = useState([]);
+	const [input, setInput] = useState('');
+
+	function handleInputChange(e) {
```

Comment:
One option would be to add the filtering of items on input change rather than the useEffect. You can create a state to track all filtered items, and based on the filter input value, filter your list based on that.

Reply:
If you do this it's important to make sure your initial state reflects all of the data passed in